x-pack/plugins/security_solution/server/lib/entity_analytics/risk_engine/risk_engine_data_client.ts

```diff
@@ -252,17 +252,23 @@ export class RiskEngineDataClient {
   public async tearDown({ taskManager, riskScoreDataClient }: TearDownParams) {
     const errors: Error[] = [];
     const addError = (e: Error) => errors.push(e);
+    const riskEngineStatus = await this.getCurrentStatus();
 
-    await removeRiskScoringTask({
-      namespace: this.options.namespace,
-      taskManager,
-      logger: this.options.logger,
-    }).catch(addError);
+    if (riskEngineStatus === RiskEngineStatusEnum.ENABLED) {
```

Comment:
I wonder if we should try and remove the task anyway and catch the error where it doesn't exist and supress it? @machadoum what do you think

Reply:
In that scenario, based on the current implementation, we receive the error message:  
"Failed to remove risk scoring task: Task not found."

This error is caught by the API, so what would be the appropriate response from the API in this case?

Option 1:
```
Status Code : 404

{
    cleanup_succesful : False ,
    errors : ["Task not found"]
}
```
OR

Option 2:

```
Status code : 200
{
    cleanup_succesful : True ,
    errors : []
}

```

IMHO, I would prefer Option 1

Reply:
I also like option one. But my biggest issue with the `if` statement is that it prevents us from calling this API a second time to recover from error scenarios.

Look consider this scenario:
1. User calls `dangerously_delete_data ` API
2. API delete risk scoring task and saved object
3. API calls `riskScoreDataClient.tearDown();`but an error happens
4. User calls `dangerously_delete_data` API again
5. It won't try to delete resources again because the engine status is uninstalled.

Reply:
Sure, thanks! I agree—the current `if` condition doesn't seem helpful for retrying in case of an error. Since we're already capturing errors and proceeding, I think it's reasonable to remove the condition for now, as it's adding more limitations than flexibility.